### ISSUE TITLE
Is there a way to set global ignore unused values? (default)

### ISSUE THREAD

**Issue by rob4629** [jaydenseric/find-unused-exports#7]
We use `export default` for a number of our functions (apparently 137 of them), and I was wondering if there was a way to set up global ignore values?

Either through the CLI: `find-unused-exports --ignore-unused default` or via a config file?

**Comment by jaydenseric** [comment]
You need to annotate the unused default exports with an ignore comment:

```js
// ignore unused exports default
```

One situation I do that is for Next.js `pages` directory modules where none of my application modules import those default exports; Next.js does.

If you are clever with your editor find and replace you can insert 137 of those ignore comments in a few minutes.

I'm against the idea of adding a setting to ignore all default exports project wide, as that would prevent unintentionally unused exports from being found.

I did think about having some sort of rules in settings, where you can ignore certain exports via globs, something like:

```json
{
  "ignore": [{
    "globs": ["pages/**/*.mjs"],
    "exports": ["default"]
  }]
}
```

But, I like the technical simplicity of the ignore comments and having used `find-unused-exports` in quite a few projects now it's worked well enough that it's hard to justify additional configuration options.

**Comment by rob4629** [comment]
Cool, I totally get the reason behind resistance to a global option... my issue is these functions _are_ being used, so I'm not sure why an error is being reported

```js
const fooBarParam = { foo: 'bar', bar: 'foo' };
export default fooBarParam;

import fooBarParam from "./getFooBarParam"

function doSomething() {
  return fooBarParam;
}
```

With the error:
```
dir/lib/getFooBarParams.js
  default
```

**Comment by jaydenseric** [comment]
You probably need to use the CLI option `--resolve-file-extensions js`, because you are not importing the modules using their full module file name including file extension.

**Comment by rob4629** [comment]
I should have added that in my initial description. We are using `--resolve-file-extensions js,ts`

**Comment by rob4629** [comment]
I've updated my comment, as it does seem to only affect functions solely used in Mocha tests

**Comment by jaydenseric** [comment]
Maybe double check that both the source and the build modules are being used. Perhaps your tests are importing the source modules, instead of the build modules? This could become an issue if you don't gitignore your build modules:

https://github.com/jaydenseric/find-unused-exports#ignoring-unused-exports

**Comment by rob4629** [comment]
Ok, I can create a new issue if you prefer... but this is for a JS/Typescript test automation framework, so there's no build modules. I've tried switching from default to named export, but it's still throwing an error for the function

I've created a function (can be called `generateFooBarParams`): 
```js
// some/dir/getFooBarParams.js
export const generateFooBarParams(x, y) => {
  return { foo: x, bar: y }
};
```

I'm using this function within a test file...
```js
// some_other/dir/x.test.js
import { expect } from 'chai';
import { generateFooBarParams } from "dir/lib/getFooBarParam";

const someVar = generateFooBarParams('a', 'b');

describe('Some set of tests', function () {
  it('uses the FooBar Params', function () {
    expect(someVar).to.have.property('foo');
  )};
)};
```


Yet despite switching to a named export/import, it still fails with
```
npx find-unused-exports --resolve-file-extensions js,ts

...
some/dir/getFooBarParams.js
  generateFooBarParams

137 unused exports in 90 modules.
```

**Comment by jaydenseric** [comment]
The unused import is in:

```
./some/dir/getFooBarParams.js
```

The import in the test module is from:

```
./some_other/dir/dir/lib/getFooBarParam.js
```

Those are different paths, so I'm not sure why you expect it to work.

**Comment by rob4629** [comment]
Ahh ok, sorry... that's likely just a typo when I was writing example the paths out (_rather than thinking about a proper structure_). I've cleared up the paths now, to be more accurate to our directory structure. 
We store our functions in a "common" sub-directory, and call them within a separate "project_X" sub-directory. We have the `package.json` file within the main `automation_framework` directory:

```
automation_framework/common/lib/getFooBarParams.js
automation_framework/project_a/tests/x.test.js
automation_framework/package.json
```

**Comment by jaydenseric** [comment]
I think the problem is that you are using a `common` bare import specifier instead of a relative path:

```js
import { generateFooBarParams } from "common/lib/getFooBarParams";
```

VS:

```js
import { generateFooBarParams } from "../../common/lib/getFooBarParams";
```

This tool only processes project modules imported from relative import paths (starting with `.`, e.g. `./` or `../`):

https://github.com/jaydenseric/find-unused-exports/blob/dd74e679e819c84e606d683f79d9ada1181a20b9/findUnusedExports.mjs#L114

If you want to have arbitrary bare import specifiers in your project instead of relative paths, you're interested in this feature:

https://github.com/jaydenseric/find-unused-exports/issues/4#issuecomment-919626516

**Comment by rob4629** [comment]
Ok, that looks like it's the issue (_switching a couple of import paths to relative results in fewer errors_). Thanks for baring with me on getting to the bottom of it.

Do you have an ETA for implementing #4?

**Comment by jaydenseric** [comment]
There is no ETA for implementing https://github.com/jaydenseric/find-unused-exports/issues/4 , because I don't have a need for that feature in any of my projects. If I get bored and have free time I might get around to it just for fun. If I suddenly need the feature for a project I might work on it sooner.

 I actually want to get this package working for Deno projects too, which kind of dampens my enthusiasm to bang out the feature in a weekend because I have to consider dependencies and things to do it that work in both environments which may or may not get complicated and require publishing other packages if nothing suitable exists.

**Comment by jaydenseric** [comment]
Support for import maps has been published in [v7.1.0](https://github.com/jaydenseric/find-unused-exports/releases/tag/v7.1.0) 🚀
